ts/features/lollipop/saga/index.ts

```diff
@@ -0,0 +1,29 @@
+import * as O from "fp-ts/lib/Option";
+import { v4 as uuid } from "uuid";
+import { put, select } from "typed-redux-saga/macro";
+import { lollipopKeyTagSelector } from "../store/reducers/lollipop";
+import { lollipopKeyTagSave } from "../store/actions/lollipop";
+import {
+  cryptoKeyGenerationSaga,
+  deletePreviousCryptoKeyPair
+} from "../../../sagas/startup/generateCryptoKeyPair";
+
+export function* generataLollipopKeySaga() {
+  const maybeOldKeyTag = yield* select(lollipopKeyTagSelector);
+  // Weather the user is logged in or not
+  // we generate a key (if no one is present)
+  // to have a key also for those users that update the app
+  // and are already logged in.
+  if (O.isNone(maybeOldKeyTag)) {
+    const newKeyTag = uuid();
+    yield* put(lollipopKeyTagSave({ keyTag: newKeyTag }));
+    yield* cryptoKeyGenerationSaga(newKeyTag, maybeOldKeyTag);
+  }
+}
+
+export function* deleteTheCurrentKeyAndGenerateNewKeyTag() {
```

Comment:
Previous method is called "generateLollipopKeySaga", should we rename this one to "deleteCurrentLollipopKeyAndGenerateNewKeyTag" for naming consistency?